### PR TITLE
Add short rationale to Java features that are not present in Kotlin

### PR DESCRIPTION
**Pull request by MarcinMoskala**
Non-private fields should be removed because we have them using @JvmField

### CODE REVIEW

**Comment by MarcinMoskala** [comment]
The reason is that when you read this page currently it looks like this are advantages of Java, not as a list of things that were intentionally removed. I often see people referencing to this list as "Kotlin has less developed generics because it does not supports Wildcard-types" or "You cannot use Primitive types in Kotlin". This is why I think there should be a short explanation what is used instead.

**Comment by MarcinMoskala** [comment]
I know that above there are already reasons why these features are not in Kotlin. But people do not read it and they just search for some cheap arguments why Java is better to later give it with info that it comes from Kotlin documentation. Adding such explanation there should help.

File: pages/docs/reference/comparison-to-java.md [hunk 1]
```diff
@@ -21,11 +21,10 @@ Kotlin fixes a series of issues that Java suffers from:
 ## What Java has that Kotlin does not
 
 * [Checked exceptions](exceptions.html)
-* [Primitive types](basic-types.html) that are not classes
-* [Static members](classes.html)
-* [Non-private fields](properties.html)
-* [Wildcard-types](generics.html)
-* [Ternary-operator `a ? b : c`](control-flow.html#if-expression)
+* [Primitive types](basic-types.html) that are not classes - are used under the hood, but cannot be used explicitely.
```
Comment (yole) [inline pages/docs/reference/comparison-to-java.md]: explicitely -> explicitly

File: pages/docs/reference/comparison-to-java.md [hunk 2]
```diff
@@ -21,11 +21,10 @@ Kotlin fixes a series of issues that Java suffers from:
 ## What Java has that Kotlin does not
 
 * [Checked exceptions](exceptions.html)
-* [Primitive types](basic-types.html) that are not classes
-* [Static members](classes.html)
-* [Non-private fields](properties.html)
-* [Wildcard-types](generics.html)
-* [Ternary-operator `a ? b : c`](control-flow.html#if-expression)
+* [Primitive types](basic-types.html) that are not classes - are used under the hood, but cannot be used explicitely. 
+* [Static members](classes.html) - replaced with [companion objects](object-declarations.html#companion-objects), and @JvmStatic.
```
Comment (yole) [inline pages/docs/reference/comparison-to-java.md]: Link to docs for `@JvmStatic?`

File: pages/docs/reference/comparison-to-java.md [hunk 3]
```diff
@@ -21,11 +21,10 @@ Kotlin fixes a series of issues that Java suffers from:
...
+* [Wildcard-types](generics.html) - replaced with declaration-site variance and type projections.
```
Comment (yole) [inline pages/docs/reference/comparison-to-java.md]: Link to docs too?

File: pages/docs/reference/comparison-to-java.md (hunk 1 above)
Comment (mikezx6r) [inline pages/docs/reference/comparison-to-java.md]: [Primitive types](basic-types.html) that are not classes. The byte-code uses primitives where possible, but they are not explicitly available.

File: pages/docs/reference/comparison-to-java.md (hunk 2 above)
Comment (mikezx6r) [inline pages/docs/reference/comparison-to-java.md]: Replaced with companion objects, top-level functions, extension functions, or @JvmStatic.

**Comment by MarcinMoskala** [comment]
I responded to all comments. Any more feedback?

**Comment by mikezx6r** [comment]
In Slack https://kotlinlang.slack.com/archives/C0922A726/p1540905520498700?thread_ts=1540898720.491000&cid=C0922A726 
there was a suggestion (I agree with it) that all of the 'Things Java has that Kotlin does not' items are actually issues or confusions in Java that Kotlin addresses. They are not things Java has that Kotlin doesn't, but rather issues Java has that Kotlin addresses.

That's a larger change, and requires more discussion, so I'm not proposing changes to your current PR, but perhaps a new one that moves these items up to the top level.

For example primitives. Java has them for performance enhancements. Kotlin doesn't need them because the compiler mostly takes care of that, thereby reducing the confusion of 'should I use primitives or classes'.

**Pull request merged**